src/main/java/com/hivemq/cli/commands/options/TlsOptions.java

```diff
@@ -76,55 +92,38 @@ public class TlsOptions {
-                        description = "The client certificate to use for client side authentication")
-    private @Nullable Collection<X509Certificate> clientCertificateChain;
-
-    @CommandLine.Option(names = {"--key"},
-                        converter = FileToPrivateKeyConverter.class,
-                        description = "The path to the client private key for client side authentication")
-    private @Nullable PrivateKey clientPrivateKey;
+    @SuppressWarnings("unused")
+    @CommandLine.Option(names = {"--ks", "--keystore"},
+                        description = "The path to the client keystore for client side authentication")
+    private @Nullable Path clientKeystore;
 
-    private boolean useBuiltSslConfig() {
-        return serverCertificateChain != null ||
-                serverCertificateChainFromDir != null ||
-                cipherSuites != null ||
-                supportedTLSVersions != null ||
-                clientPrivateKey != null ||
-                clientCertificateChain != null ||
-                useTls;
-    }
+    @SuppressWarnings("unused")
+    @CommandLine.Option(names = {"--kspw", "--keystore-password"}, description = "The password for the keystore")
+    private @Nullable String clientKeystorePassword;
 
-    public @Nullable MqttClientSslConfig buildSslConfig() throws Exception {
-        setDefaultOptions();
+    @SuppressWarnings("unused")
+    @CommandLine.Option(names = {"--kspkpw", "--keystore-private-key-password"},
+                        description = "The password for the private key inside the keystore")
+    private @Nullable String clientKeystorePrivateKeyPassword;
 
-        if (!useBuiltSslConfig()) {
-            return null;
-        }
+    @SuppressWarnings("unused")
+    @CommandLine.Option(names = {"--ts", "--truststore"},
+                        description = "The path to the client truststore to enable encrypted certificate based communication")
+    private @Nullable Path clientTruststore;
 
-        if (serverCertificateChainFromDir != null) {
-            if (serverCertificateChain == null) {
-                serverCertificateChain = serverCertificateChainFromDir;
-            } else {
-                serverCertificateChain.addAll(serverCertificateChainFromDir);
-            }
-        }
+    @SuppressWarnings("unused")
+    @CommandLine.Option(names = {"--tspw", "--truststore-password"},
+                        description = "The path to the client truststore for client side authentication")
+    private @Nullable String clientTruststorePassword;
 
-        // build trustManagerFactory for server side authentication and to enable tls
-        TrustManagerFactory trustManagerFactory = null;
-        if (serverCertificateChain != null && !serverCertificateChain.isEmpty()) {
-            trustManagerFactory = buildTrustManagerFactory(serverCertificateChain);
+    public @Nullable MqttClientSslConfig buildSslConfig() throws Exception {
```

Comment:
I think we need to put some programmatic validation in place, to avoid cases where e.g. both a client-cert and a client-keystore is defined.